chain/chain/src/resharding.rs

```diff
@@ -39,12 +42,77 @@ pub struct StateSplitResponse {
     pub new_state_roots: Result<HashMap<ShardUId, StateRoot>, Error>,
 }
 
+fn get_checked_account_id_to_shard_uid_fn(
+    shard_uid: ShardUId,
+    new_shards: Vec<ShardUId>,
+    next_epoch_shard_layout: ShardLayout,
+) -> impl Fn(&AccountId) -> ShardUId {
+    let split_shard_ids: HashSet<_> = new_shards.into_iter().collect();
+    move |account_id: &AccountId| {
+        let new_shard_uid = account_id_to_shard_uid(account_id, &next_epoch_shard_layout);
+        // check that all accounts in the shard are mapped the shards that this shard will split
+        // to according to shard layout
+        assert!(
+            split_shard_ids.contains(&new_shard_uid),
+            "Inconsistent shard_layout specs. Account {:?} in shard {:?} and in shard {:?}, but the former is not parent shard for the latter",
+            account_id,
+            shard_uid,
+            new_shard_uid,
+        );
+        new_shard_uid
+    }
+}
+
+// Return iterate over flat storage to get key, value. Used later in the get_trie_update_batch function.
+fn get_flat_storage_iter<'a>(
+    store: &'a Store,
+    shard_uid: ShardUId,
+) -> impl Iterator<Item = (Vec<u8>, Vec<u8>)> + 'a {
+    let trie_storage = TrieDBStorage::new(store.clone(), shard_uid);
+    store_helper::iter_flat_state_entries(shard_uid, &store, None, None).map(
```

Comment:
please add the TODO that this needs to be adjusted to use the flat state snapshot at correct block